src/iuse.cpp

```diff
@@ -8068,6 +8068,36 @@ int iuse::hairkit(player *p, item *it, bool, const tripoint&)
     return it->type->charges_to_use();
 }
 
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need soap to use this." ) );
+    } else if ( g->m.has_nearby_water( g->u.pos() ) ) {
+        int pos = g->inv_for_filter( _( "Wash what?" ), []( const item & itm ) {
+            return itm.has_flag( "FILTHY" ); } );
+    item *mod = &( p->i_at( pos ) );
+
+    if( mod == NULL || mod->is_null() ) {
```

Comment:
`mod == NULL` check is excessive. `i_at()` shall never return `NULL`, it returns a dummy item instead (`mod->is_null()` checks this).

Also, use `nullptr` everywhere. `NULL` is simply a macro which is replaced with `0` by preprocessors. `NULL` is a marker of old code.